java/client/src/main/java/glide/api/models/commands/SetOptions.java

```diff
@@ -0,0 +1,129 @@
+    private final TimeToLive expiry;
+
+    /** Conditions which define whether new value should be set or not. */
+    @RequiredArgsConstructor
+    @Getter
+    public enum ConditionalSet {
+        /**
+         * Only set the key if it does not already exist. Equivalent to <code>XX</code> in the Redis
+         * API.
+         */
+        ONLY_IF_EXISTS("XX"),
+        /** Only set the key if it already exists. Equivalent to <code>NX</code> in the Redis API. */
+        ONLY_IF_DOES_NOT_EXIST("NX");
+
+        private final String redisApi;
+    }
+
+    /** Configuration of value lifetime. */
+    @Builder
+    public static final class TimeToLive {
+        /** Expiry type for the time to live */
+        @NonNull private TimeToLiveType type;
+
+        /**
+         * The amount of time to live before the key expires. Ignored when {@link
+         * TimeToLiveType#KEEP_EXISTING} type is set.
+         */
+        private Integer count;
+    }
+
+    /** Types of value expiration configuration. */
+    @RequiredArgsConstructor
+    @Getter
+    public enum TimeToLiveType {
+        /**
+         * Retain the time to live associated with the key. Equivalent to <code>KEEPTTL</code> in the
+         * Redis API.
+         */
+        KEEP_EXISTING("KEEPTTL"),
+        /**
+         * Set the specified expire time, in seconds. Equivalent to <code>EX</code> in the Redis API.
+         */
+        SECONDS("EX"),
+        /**
+         * Set the specified expire time, in milliseconds. Equivalent to <code>PX</code> in the Redis
+         * API.
+         */
+        MILLISECONDS("PX"),
+        /**
+         * Set the specified Unix time at which the key will expire, in seconds. Equivalent to <code>
+         * EXAT</code> in the Redis API.
+         */
+        UNIX_SECONDS("EXAT"),
+        /**
+         * Set the specified Unix time at which the key will expire, in milliseconds. Equivalent to
+         * <code>PXAT</code> in the Redis API.
+         */
+        UNIX_MILLISECONDS("PXAT");
+
+        private final String redisApi;
+    }
+
+    /** String representation of {@link #returnOldValue} when set. */
+    public static final String RETURN_OLD_VALUE = "GET";
+
+    /**
+     * Converts SetOptions into a String[] to add to a {@link Command} arguments.
+     *
+     * @return String[]
+     */
+    public String[] toArgs() {
+        List<String> optionArgs = new ArrayList<>();
+        if (conditionalSet != null) {
+            optionArgs.add(conditionalSet.redisApi);
+        }
+
+        if (returnOldValue) {
+            optionArgs.add(RETURN_OLD_VALUE);
+        }
+
+        if (expiry != null) {
+            optionArgs.add(expiry.type.redisApi);
+            if (expiry.type != TimeToLiveType.KEEP_EXISTING) {
+                if (expiry.count == null) {
+                    throw new RequestException(
+                            "Set command received expiry type " + expiry.type + ", but count was not set.");
```

Comment:
but count was not set => but no value was set

Reply:
I'd prefer to keep `count`, which is consistent with Node client.